apps/gallery/test/marionette/lib/gallery.js

```diff
@@ -310,6 +386,45 @@ Gallery.prototype = {
     }.bind(this));
   },
 
+  switchToGalleryFrame: function() {
+    var galleryFrame = this.client.findElement('iframe[src*="' +
+                                          Gallery.ORIGIN + '"]');
+    this.waitFor(galleryFrame);
+    this.client.switchToFrame(galleryFrame);
+  },
+
+  tapFirstThumbnail: function() {
+    this.waitFor(Gallery.Selector.thumbnail);
+    var thumbnails = this.thumbnails;
+    this.waitFor(thumbnails[0]);
+    thumbnails[0].click();
+  },
+
+  selectImage: function() {
+    this.switchToGalleryFrame();
+    this.tapFirstThumbnail();
+  },
+
+  shareImage: function(appName) {
+    this.waitFor(Gallery.Selector.shareButton).tap();
+    this.menuOptionButton(appName).tap();
+  },
```

Comment:
This function also involves the gallery and systme apps, so maybe it should not be in this gallery-specific library.

Can you move it to the test itself and write something like this?

```
gallery.shareButton.tap();
system.activityMenuButton(appName).tap()
```